### ISSUE TITLE
AdaptiveMetropolis and HDF5 with multiple chains

### ISSUE THREAD

**Issue by mpjdem** [pymc-devs/pymc3#302]
PyMC 2.2, Numpy 1.7.1 and pyTables 2.4.0

When using the HDF5 back-end together with multiple-chain AdaptiveMetropolis, stats() cannot be generated with 'chain=None'. Inspecting the H5 file with vitables, stochastic parameter values are zero-dimensional in chain0, but of shape (1,) in the next chains. As a result, they cannot be stacked to generate summary statistics across all chains. Other functions combining traces of different chains will similarly fail.

The reason for this appears to be the AdaptiveMetropolis propose() function, which adds jump values of shape (1,)  to the zero-dimensional stochastic value. Since the chain0 table is created before the first jump value is added, the column type is correct in this case; but for every next table, the column values are of shape (1,).

Take this example:

``` python
import pymc as mc
import numpy as np

dat = np.random.rand(1000)
mu = mc.Normal('mu', 0.5, 1e6, value=0.5)
tau = mc.Uniform('tau', 1e-6, 1e2, value=1e-6)
obs = mc.Normal('obs', mu, tau, value=dat, observed=True)

m = mc.MCMC([mu, tau, obs], db='hdf5', dbname='tmp.h5')
m.use_step_method(mc.AdaptiveMetropolis, [mu, tau])

m.sample(100, progress_bar=False)
m.sample(100, progress_bar=False)

print mu.stats(chain=0), '\n'
print mu.stats(chain=1), '\n'
print mu.stats(chain=None), '\n'

m.db.close()
```

The third stats() call fails. Delete the database file however, comment out the AdaptiveMetropolis line, and it will work.

**Comment by fonnesbeck** [comment]
This should be fixed in 38055d2. Please test.

**Comment by mpjdem** [comment]
That works, thanks.
